odl/discr/grid.py

```diff
@@ -246,72 +223,81 @@ def max_pt(self):
         """
         return np.array([vec[-1] for vec in self.coord_vectors])
 
+    # min, max and extent are for set duck-typing
     def min(self):
-        """Return vector with minimal cell coordinates per axis.
+        """Return `min_pt`.
 
-        This is relevant if the grid was initialized with
-        `as_midp` ``True``, in which case the minimum is half a cell
-        smaller than the minimum grid point.
+        See also
+        --------
+        Intervalprod.min
 
         Examples
         --------
-        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2], as_midp=False)
+        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2])
         >>> g.min()
         array([ 1., -2.])
-        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2], as_midp=True)
-        >>> g.min()
-        array([ 0.5 , -3.75])
         """
-        if not self.as_midp:
-            return self.min_pt
-        elif self._exact_min is not None:
-            return self._exact_min
-        else:
-            minpt_cell_size = np.array([cs[0] for cs in self.cell_sizes()])
-            return self.min_pt - minpt_cell_size / 2
+        return self.min_pt
 
     def max(self):
-        """Return vector with maximal cell coordinates per axis.
+        """Return `max_pt`.
 
-        This is relevant if the grid was initialized with
-        `as_midp` ``True``, in which case the maximum is half a cell
-        larger than the maximum grid point.
+        See also
+        --------
+        Intervalprod.max
 
         Examples
         --------
-        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2], as_midp=False)
+        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2])
         >>> g.max()
         array([ 5.,  2.])
-        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2], as_midp=True)
-        >>> g.max()
-        array([ 6.5 ,  2.25])
         """
-        if not self.as_midp:
-            return self.max_pt
-        elif self._exact_max is not None:
-            return self._exact_max
-        else:
-            maxpt_cell_size = np.array([cs[-1] for cs in self.cell_sizes()])
-            return self.max_pt + maxpt_cell_size / 2
-
-    # Methods
+        return self.max_pt
 
     def extent(self):
-        """Return a vector containing the total grid extent."""
-        return self.max() - self.min()
+        """Return a vector containing the total grid extent.
+
+        Examples
+        --------
+        >>> g = TensorGrid([1, 2, 5], [-2, 1.5, 2])
+        >>> g.extent()
+        array([ 4.,  4.])
+        """
+        return self.max_pt - self.min_pt
+
+    def convex_hull(self):
+        """The smallest `IntervalProd` containing this grid.
```

Comment:
Related to above, this is imo, a perfect example of a good docstring, short and to the point.